File: Minimum Exchange.java
class Solution
{
    public int MinimumExchange(char[][] matrix) {
        int a = 0, b = 0;
        int m = matrix.length, n = matrix[0].length;
        for(int i = 0; i < m; ++i){
            for(int j = 0; j < n; ++j){
                if(matrix[i][j] == 'A'){
                    a++;
                } else {
                    b++;
                }
            }
        }
        if(a > b){
            return Count('A', 'B', matrix);
        } else if(a < b){
            return Count('B', 'A', matrix) ;
        } else {
            return Math.min(Count('A', 'B', matrix), Count('B', 'A', matrix));
        }
    }

    private int Count(char x, char y, char[][] matrix) {
        int m = matrix.length, n = matrix[0].length;
        int cnt = 0;
        for(int i = 0; i < m; ++i){
            char ch = x;
            if(i % 2 == 1){
                ch = y;
            }
            for(int j = 0; j < n; ++j){
                if(matrix[i][j] != ch){
                    cnt++;
                }
                if(ch == x){
                    ch = y;
                } else {
                    ch = x;
                }
            }
        }
        return cnt / 2;
    }
}

File: Minimum Exchange GFG/MinimumExchange.cpp
class Solution {
public:
int MinimumExchange(vector<vector<char>>matrix){
    // Code here
    int ans1=0;
    int ans2=0;
    for(int i=0;i<matrix.size();i++){
        for(int j=0;j<matrix[0].size();j++){
            
             if((i+j)%2==1 && matrix[i][j]=='A'){
                ans1++;
            }
            if((i+j)%2==0 && matrix[i][j]=='B'){
                ans1++;
            }
           
        }
    }
    
      for(int i=0;i<matrix.size();i++){
        for(int j=0;j<matrix[0].size();j++){
            if((i+j)%2==0 && matrix[i][j]=='A'){
                ans2++;
            }
            if((i+j)%2==1 && matrix[i][j]=='B'){
                ans2++;
            }
        }
    }
    return min(ans1,ans2);
}

};

//  More concise code
 
//  int MinimumExchange(vector<vector<char>>matrix){
//     // Code here
//     int m = matrix.size(), n = matrix[0].size();
//     int cnt = 0;
    
//     for(int i=0; i<m; i++)
//     {
//         for(int j=0; j<n; j++)
//         {
//             if((i+j)%2 == 0 and matrix[i][j] == 'B')
//             cnt++;
            
//             if((i+j)%2 and matrix[i][j] == 'A')
//             cnt++;
//         }
//     }
    
//     return min(cnt, m*n - cnt);
// }
